pydicom/dataelem.py

```diff
@@ -811,20 +811,19 @@ def DataElement_from_raw(
                     VR = 'UN'
                     msg += " - setting VR to 'UN'"
                     warnings.warn(msg)
-    elif VR == 'UN' and config.replace_un_with_known_vr:
-        # handle rare case of incorrectly set 'UN' in explicit encoding
-        # see also DataElement.__init__()
-        if raw.tag.is_private:
-            VR = _private_vr_for_tag(dataset, raw.tag)
-        elif (
-            raw.length == 0xffffffff
-            or raw.value is None
-            or len(raw.value) < 0xffff
-        ):
-            try:
-                VR = dictionary_VR(raw.tag)
-            except KeyError:
-                pass
+    elif VR == 'UN':
+        if raw.length == 0xffffffff:
+            VR = 'SQ'
```

Comment:
I think this makes sense, but I couldn't get a test together to actually hit this. Maybe it is not a real use case...

Reply:
It wasn't - the check was placed in the wrong location, fixed now.